Subject: [PATCH] gitweb: correct quot_cec function

If a control character is < 0x10, make sure the hex value shown uses
a leading 0 after the \ instead of a space.  Also make sure the 'x'
following the \ is shown to indicate it's a hexadecimal value.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>
---
gitweb/gitweb.perl | 2 +-
1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gitweb/gitweb.perl b/gitweb/gitweb.perl
index 937177c3..8ccc7e20 100755
--- a/gitweb/gitweb.perl
+++ b/gitweb/gitweb.perl
@@ -1765,7 +1765,7 @@ sub quot_cec {
 	);
 	my $chr = ( (exists $es{$cntrl})
 		    ? $es{$cntrl}
-		    : sprintf('\%2x', ord($cntrl)) );
+		    : sprintf('\x%02x', ord($cntrl)) );
 	if ($opts{-nohtml}) {
 		return $chr;
 	} else {
---
